vocab-builder-core/src/main/java/com/github/tkurz/sesame/vocab/GenerationSetting.java

```diff
@@ -0,0 +1,53 @@
+package com.github.tkurz.sesame.vocab;
+
+import com.google.common.base.CaseFormat;
+
+public class GenerationSetting {
+
+	private String caseFormat;
+	private String constantPrefix;
+	private String constantSuffix;
+	public CaseFormat getCaseFormat() {
+		try {
+			if ( caseFormat != null ) {
+				return CaseFormat.valueOf(caseFormat);
+			}
+		} catch (Exception e) {
+		}
+		return null;
+	}
+	public void setCaseFormat(String caseFormat) {
+		this.caseFormat = caseFormat;
+	}
+	public String getConstantPrefix() {
+		return constantPrefix;
+	}
+	public void setConstantPrefix(String constantPrefix) {
+		this.constantPrefix = constantPrefix;
+	}
+	public String getConstantSuffix() {
+		return constantSuffix;
+	}
+	public void setConstantSuffix(String constantSuffix) {
+		this.constantSuffix = constantSuffix;
+	}
+	public static GenerationSetting createDefault(CaseFormat format, String prefix, String suffix) {
+		GenerationSetting setting = new GenerationSetting();
+		setting.caseFormat = (format != null ? format.name() : null);
+		setting.constantPrefix = prefix;
+		setting.constantSuffix = suffix;
+		return setting;
+		
+	}
+	
+	public GenerationSetting mergeWith(GenerationSetting other) {
```

Comment:
Need javadoc to specify the contract here, as it isn't clear what semantics the merge operation has without reading the code.